velox/connectors/hive/tests/HivePartitionFunctionTest.cpp

```diff
@@ -83,3 +83,101 @@ TEST_F(HivePartitionFunctionTest, bool) {
   assertPartitions(values, 500, {0, 1, 0, 0, 1});
   assertPartitions(values, 997, {0, 1, 0, 0, 1});
 }
+
+TEST_F(HivePartitionFunctionTest, tinyint) {
+  auto values = vm_.flatVectorNullable<int8_t>(
+      {std::nullopt,
+       64,
+       std::numeric_limits<int8_t>::min(),
+       std::numeric_limits<int8_t>::max()});
+
+  assertPartitions(values, 1, {0, 0, 0, 0});
+  assertPartitions(values, 2, {0, 0, 0, 1});
+  assertPartitions(values, 500, {0, 64, 20, 127});
+  assertPartitions(values, 997, {0, 64, 355, 127});
+}
+
+TEST_F(HivePartitionFunctionTest, smallint) {
+  auto values = vm_.flatVectorNullable<int16_t>(
+      {std::nullopt,
+       30'000,
+       std::numeric_limits<int16_t>::min(),
+       std::numeric_limits<int16_t>::max()});
+
+  assertPartitions(values, 1, {0, 0, 0, 0});
+  assertPartitions(values, 2, {0, 0, 0, 1});
+  assertPartitions(values, 500, {0, 0, 380, 267});
+  assertPartitions(values, 997, {0, 90, 616, 863});
+}
+
+TEST_F(HivePartitionFunctionTest, integer) {
+  auto values = vm_.flatVectorNullable<int32_t>(
+      {std::nullopt,
+       2'000'000'000,
+       std::numeric_limits<int32_t>::min(),
+       std::numeric_limits<int32_t>::max()});
+
+  assertPartitions(values, 1, {0, 0, 0, 0});
+  assertPartitions(values, 2, {0, 0, 0, 1});
+  assertPartitions(values, 500, {0, 0, 0, 147});
+  assertPartitions(values, 997, {0, 54, 0, 482});
+}
+
+TEST_F(HivePartitionFunctionTest, real) {
+  auto values = vm_.flatVectorNullable<float>(
+      {std::nullopt,
+       2'000'000'000.0,
+       std::numeric_limits<float>::min(),
```

Comment:
Use `numeric_limits<float>::lowest()`

Reply:
Fixed, didn't noticed the difference between min vs lowest, thanks for suggestion!